eng/tools/versioning/set-dev.js

```diff
@@ -23,55 +23,145 @@ const path = require("path");
+        parsedDepMinVersion.minor == parsedPackageVersion.minor &&
+        parsedDepMinVersion.patch == parsedPackageVersion.patch
+      ) {
+        rushPackages[
+          depName
+        ].newVer = `${parsedPackageVersion.major}.${parsedPackageVersion.minor}.${parsedPackageVersion.patch}-dev.${buildId}`;
+        dependencySection[
+          depName
+        ] = `^${parsedPackageVersion.major}.${parsedPackageVersion.minor}.${parsedPackageVersion.patch}-dev`;
+      }
+    }
+  }
+  return rushPackages;
+};
+
+const updateInternalDependencyVersions = (rushPackages, package, buildId) => {
+  console.log(package);
+  console.log("checking dependencies ..");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.dependencies,
+    buildId
   );
 
-  console.log(appropriateVersionedPackages);
+  console.log("checking devDependencies ..");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.devDependencies,
+    buildId
+  );
 
-  let targetPackages = appropriateVersionedPackages.map(project =>
-    path.resolve(path.join(repoRoot, project.projectFolder))
+  console.log("checking peerDependencies ..");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.peerDependencies,
+    buildId
   );
 
-  for (const targetFolder of targetPackages) {
-    const targetPackageJson = path.join(targetFolder, "package.json");
-    const packageJsonContents = await versionUtils.readFileJson(
-      targetPackageJson
-    );
-    const newVersion = getDevVersion(packageJsonContents.version, buildId);
-    console.log(`File ${targetPackageJson} version updated to ${newVersion}`);
-    const updatedPackageSpec = {
-      ...packageJsonContents,
-      version: newVersion
-    };
-    await versionUtils.writePackageJson(targetPackageJson, updatedPackageSpec);
-
-    await versionUtils.updatePackageConstants(
-      targetFolder,
-      packageJsonContents,
-      newVersion
-    );
+  return rushPackages;
+};
+
+async function main(argv) {
+  const buildId = argv["build-id"];
+  const repoRoot = argv["repo-root"];
+  const service = argv["service"];
+
+  var rushPackages = await versionUtils.getRushPackageJsons(repoRoot);
+
+  // Set all the new versions
+  console.log(`Updating packages with build ID ${buildId}`);
+  for (const package of Object.keys(rushPackages)) {
+    if (
+      (rushPackages[package].versionPolicy == "client" ||
+        rushPackages[package].versionPolicy == "core") &&
+      rushPackages[package].projectFolder.startsWith(`sdk/${service}`)
+    ) {
+      console.log("package updated = ");
+      console.log(package);
+      rushPackages = updatePackageVersion(rushPackages, package, buildId);
+      console.log(rushPackages[package].newVer);
+    }
+  }
+
+  // Update any references to internal projects with the new versions
+  for (const package of Object.keys(rushPackages)) {
+    if (
```

Comment:
This filter is being executed twice. We can execute it once saving lines of code and making it so that future logic changes only have to edit one version of the filter logic. 

```
let targetPackages = [];
for (const package of Object.keys(rushPackages)) { 
    if (
      ["client", "core"].includes(rushPackages[package].versionPolicy)
      && rushPackages[package].projectFolder.startsWith(`sdk/${service}`)
    ) {
      targetPackages.push(package);
    }
}

for (const package of targetPackages)) {
    console.log("package updated = ");
    console.log(package);
    rushPackages = updatePackageVersion(rushPackages, package, buildId);
    console.log(rushPackages[package].newVer);
}

...
```